src/main/scala/viper/gobra/ast/internal/transform/CGEdgesTerminationTransform.scala

```diff
@@ -0,0 +1,198 @@
+                  val src = m.getMeta
+                  val assumeFalse = in.Assume(in.ExprAssertion(in.BoolLit(b = false)(src))(src))(src)
+                  val newBody = {
+                    in.Block(
+                      decls = Vector.empty,
+                      stmts = assumeFalse +: implementations.toVector.flatMap { t: in.Type =>
+                        table.lookup(t, proxy.name).map {
+                          case implProxy: in.MethodProxy =>
+                            in.If(
+                              in.EqCmp(in.TypeOf(m.receiver)(src), typeAsExpr(t)(src))(src),
+                              in.Seqn(Vector(in.MethodCall(m.results map parameterAsLocalValVar, in.Parameter.In(m.receiver.id, t)(src), implProxy, m.args)(src), in.Return()(src)))(src),
+                              in.Seqn(Vector())(src)
+                            )(src)
+                          case v => Violation.violation(s"Expected a MethodProxy but got $v instead.")
+                        }
+                      }
+                    )(src)
+                  }
+                  val newMember = in.Method(m.receiver, m.name, m.args, m.results, m.pres, m.posts, m.terminationMeasures, Some(newBody))(src)
+                  methodsToRemove += m
+                  methodsToAdd += newMember
+                  definedMethodsDelta += proxy -> newMember
+
+                /**
+                  * Transforms the abstract pure methods from interface declarations into non-abstract pure methods containing calls
+                  * to all implementations' corresponding methods. The new body has the form
+                  *   {
+                  *      true?
+                  *       call fallbackFunction on recv :
+                  *       (typeOf(recv) == impl1 ? call method from impl1 on recv :
+                  *         (typeOf(recv) == impl2 ? call method from impl2 on recv :
+                  *           (...  :
+                  *             typeOf(recv) == implN ? call implementation method from implN on recv : call fallbackFunction on recv)))
+                  *   }
+                  *
+                  *   This transformation generates a fallbackFunction, an abstract function which receives the receiver and parameters
+                  *   of the original method and has the same return type and spec. For the pure method
+                  *     requires [PRE]
+                  *     decreases [MEASURE]
+                  *     pure func (r recv) M (x1 T1, ..., xN TN) (res TRes)
+                  *
+                  *   we generate the following fallback:
+                  *     requires [PRE]
+                  *     ensures res == r.N(x1, ..., xN)
+                  *     decreases _
+                  *     pure func (r recv) M_fallback(x1 T1, ... xN TN) (res TRes)
+                  *
+                  *   Notice that the postcondition `res == r.N(x1, ..., xN)` is required, because the interface encoding
```

Comment:
What is `N`? Should this be `M` instead? If so, I'm surprised that it's working as the dependencies seem to be circular: M calls M_fallback whose specification relies on M's specification...